frontend/morpheus/App.js

```diff
@@ -0,0 +1,360 @@
+import React, { useEffect, useState } from "react";
+import { withRouter } from "react-router-dom";
+import { connect } from "react-redux";
+import { useSnackbar } from "notistack";
+import PropTypes from "prop-types";
+import axios from "axios";
+import debounce from "lodash.debounce";
+
+import Loading from "../components/Loading";
+import PageLayout from "../components/PageLayout";
+import MenuUsers from "../components/MenuUsers";
+import InviteToMeetingDialog from "../components/InviteToMeetingDialog";
+import ReceiveInviteDialog from "../components/ReceiveInviteDialog";
+import MessageDialog from "../components/MessageDialog";
+import SnackbarActions from "../components/SnackbarActions";
+import Error500 from "../components/Error500";
+import PageRoutes, { AppBarRouter } from "./Routes";
+import { showBrowserNotification } from "../notification";
+import {
+  initProfile,
+  initEvents,
+  closeConnection,
+  getCurrentUser,
+  emitEnterInRoom,
+  emitInviteUser,
+  getCurrentRoomId
+} from "./socket";
+import {
+  setCurrentUser,
+  setCurrentRoom,
+  addRooms,
+  syncOffice,
+  changeUsersFilter,
+  addUser,
+  addError,
+  removeUser,
+  userEnterMeeting,
+  userLeftMeeting
+} from "./store/actions";
+import {
+  selectRooms,
+  selectCurrentUser,
+  selectUsers,
+  selectUsersFilter,
+  selectCurrentRoom,
+  selectError,
+  selectSettings
+} from "./store/selectors";
+
+const useSocket = (
+  toggleLoading,
+  setLoggedIn,
+  onSetCurrentUser,
+  onSetCurrentRoom,
+  onAddRooms,
+  onAddError
+) => {
+  useEffect(() => {
+    const profile = initProfile();
+
+    if (profile.isProfileStored()) {
+      onSetCurrentUser(getCurrentUser());
```

Comment:
The `useSocket` and `useEvents` are custom [React Hooks](https://reactjs.org/docs/hooks-intro.html) and not HOC. But it makes perfect sense to separate them into files and I did it.

The organization of this hooks it's a little confusing, I agree. I started developing the logic inside the useEffect, it got bigger, so I separated in a custom hook, and after it was bigger I saw that it was able to separate into two different custom hooks. But there's still a lot of behavior coupled with them.

I've tried to separate the maximum of chat logic in `socket.js`, but as **Morpheus is a proposal**, I tried to do as little as possible in what already existed, in case you did not like it would be easy after removing it. So I ended up leaving some technical debts.

Since you all like Morpheus, we can see a better way to organize it when it's replacing the old front end.